internal/gengapic/gengapic.go

```diff
@@ -297,6 +297,13 @@ func buildAccessor(field string) string {
 func (g *generator) lookupField(msgName, field string) *descriptor.FieldDescriptorProto {
 	var desc *descriptor.FieldDescriptorProto
 	msg := g.descInfo.Type[msgName]
+
+	// It could be that "msgName" is a field that is not a message.
+	// Fail cleanly in that case.
+	if msg == nil {
+		return desc
+	}
```

Comment:
I don't understand this case, can you share an example? The `msgName` is supposed to be the fully-qualified name of the Message to look up and then find the field in.

Reply:
This is a minor workaround for a specific case I ran into during my fixes to `queryParams`.
In `queryParams`, I want to get the field pointed to by the `body` parameter and remove all of its fields from the set of query params.
If the `body` parameter points to a field in the request that is a primitive type, it's cleaner to just call `lookupField` and let it handle the lookup failure instead of changing the logic in `queryParams`, and it adds a little bit of error checking.

If you would prefer this function not be augmented so I'll add a comment about the preconditions and change the logic in `queryParams`. I'd like to keep it like this, though, if possible.

Reply:
Nah that's fine, but maybe change the comment to be more generic like "if the msg doesn't exist ..."